src/analysis/utils/runtime-utils.ts

```diff
@@ -104,6 +117,23 @@ export const trimRuntimesOldestFirst = (runtimes: Runtime[]): Runtime[] => {
   return sortedRuntimes;
 };
 
+/**
+ * The first Jupyter or RStudio image URL on the provided GCP runtime object.
+ */
+export const getImageUrlFromRuntime = (
+  runtimeDetails: Pick<GetRuntimeItem, 'runtimeImages'> | undefined
+): string | undefined => {
+  const images: LeoRuntimeImage[] = runtimeDetails?.runtimeImages ?? [];
+  const gcpToolLabels: RuntimeToolLabel[] = cloudRuntimeTools.GCP.map(({ label }) => label);
+  return images.find(({ imageType }) => {
```

Comment:
may use an intermediate variable for the find and return `x?.imageUrl` here for a bit more readability